Dockerfile

```diff
@@ -30,8 +30,10 @@ FROM python:3.10.9-slim-bullseye
 # update and install dependencies
 RUN apt update
 RUN apt upgrade -y
-# copy and install wheel
+# copy and install requirements and wheel
 WORKDIR /service
+COPY --from=builder /service/requirements.txt /service
+RUN pip install --no-deps -r requirements.txt
 COPY --from=builder /service/dist/ /service
 RUN pip install *.whl
```

Comment:
Should there be a `--no-deps` here as a safeguard so that at least we get a non-working image when the requirements.txt is out of sync / missing a dep rather than silent pip-style dependency resolution for missing deps? 🤔

Reply:
We use requirements.txt as a lock-file. It is created automatically with pip-compile and should contain all dependencies already. If any dependencies of the services would still be missing (should never happen), then they would be installed in the next step where we install the wheel.

One problem that still can occur is when some requirements are not listed in the pyproject.toml, but we don't notice it during development if they are installed (maybe indirectly) as part of the dev-requirements.

Reply:
Sorry, misunderstood your question. Agree, for sake of reproducibility and security, it may be better to also use "--no-deps".

Reply:
Will create a new PR for that later.